nestor_api/lib/docker.py

```diff
@@ -49,30 +41,26 @@ def build(app_name, context):
     return image_tag
 
 
-def has_docker_image(app_name, tag):
+def has_docker_image(app_name: str, tag: str) -> bool:
     """Checks if the docker image already exists for a given app and tag"""
     stdout = io.execute(f"docker images {app_name}:{tag} --quiet")
     return len(stdout) != 0
 
 
-def get_registry_image_tag(app_name, image_tag, registry):
+def get_registry_image_tag(app_name: str, image_tag: str, registry: dict) -> str:
     """Returns the image name for a given organization, app and tag"""
     return f"{registry['organization']}/{app_name}:{image_tag}"
 
 
-def push(app_name, repository):
+def push(app_name: str, image_tag: str, app_config: dict) -> None:
     """Push an image to the configured docker registry"""
-
-    # This will need to be done a bit differently to work with GCP registry
-
-    app_config = config.get_app_config(app_name)
-
-    image_tag = git.get_last_tag(repository)
-
     if not has_docker_image(app_name, image_tag):
         raise RuntimeError("Docker image not available")
 
-    registry = app_config["docker"]["registry"]
+    # This will need to be done a bit differently to work with other registries (GCP)
+    # -> the config schema currently expects
+    #   {docker: {registries: {[name: string]: {id: string, organization: string}[]}}}
+    registry = app_config["docker"]["registries"]["docker.com"][0]
```

Comment:
Are we sure all those keys will always be in the dict? And that it would always have at least one value in the array?

I maybe would have the tendency to use the following to avoid raising a `KeyError` if something is missing
```python
registries = app_config.get("docker", {}).get("registries", {}).get("docker.com")
if registries is None: # we might even check for len() here, not sure it is 100% necessary, though
    logger.error("Registries not found in app_config", app_config)
    raise Exception("blabla")

registry = registries[0]
```

What do you think?

Reply:
I wanted to discuss if we could simplify this field as it looks a bit too complex compared to what we need. I added a note in the project's board 🙂